components/radio/index.css

```
@@ -20,471 +20,412 @@ governing permissions and limitations under the License.
-                          var(--spectrum-radio-button-border-color-default)));
-    border-style: solid;
-
-    transition: border var(--mod-radio-animation-duration,
-                          var(--spectrum-radio-animation-duration)) ease-in-out,
-                box-shadow var(--mod-radio-animation-duration,
-                              var(--spectrum-radio-animation-duration)) ease-in-out;
-  }
-
-  /* focus ring is not focused state */
-  &::after {
-    border-radius: 50%;
-
-    content: "";
-    display: block;
-    position: absolute;
-    inset-inline-start: 50%;
-    inset-block-start: 50%;
-    transform: translateX(-50%) translateY(-50%);
-
-    transition: opacity var(--mod-radio-animation-duration,
-                            var(--spectrum-radio-animation-duration)) ease-out,
-                margin var(--mod-radio-animation-duration,
-                          var(--spectrum-radio-animation-duration)) ease-out;
-
-    &:dir(rtl) {
-      transform: translateX(50%) translateY(-50%);
-    }
-  }
+	position: relative;
+	box-sizing: border-box;
+	inline-size: var(--mod-radio-button-control-size, var(--spectrum-radio-button-control-size));
+	block-size: var(--mod-radio-button-control-size, var(--spectrum-radio-button-control-size));
+	margin-block-start: var(--mod-radio-button-top-to-control, var(--spectrum-radio-button-top-to-control));
+
+	flex-grow: 0;
+	flex-shrink: 0;
+
+	/* selection indicator unchecked */
+	&::before {
+		display: block;
+		z-index: 0;
+		content: "";
+		box-sizing: border-box;
+		position: absolute;
+
+		inline-size: var(--mod-radio-button-control-size, var(--spectrum-radio-button-control-size));
+		block-size: var(--mod-radio-button-control-size, var(--spectrum-radio-button-control-size));
+
+		background-color: var(--highcontrast-radio-button-background-color, var(--mod-radio-button-background-color, var(--spectrum-radio-button-background-color)));
+
+		border-radius: 50%;
+		border-width: var(--mod-radio-border-width, var(--spectrum-radio-border-width));
+		border-color: var(--highcontrast-radio-button-border-color-default, var(--mod-radio-button-border-color-default, var(--spectrum-radio-button-border-color-default)));
+		border-style: solid;
+
+		transition:
+			border var(--mod-radio-animation-duration, var(--spectrum-radio-animation-duration)) ease-in-out,
+			box-shadow var(--mod-radio-animation-duration, var(--spectrum-radio-animation-duration)) ease-in-out;
+	}
+
+	/* focus ring is not focused state */
+	&::after {
+		border-radius: 50%;
+
+		content: "";
+		display: block;
+		position: absolute;
+		inset-inline-start: 50%;
+		inset-block-start: 50%;
+		transform: translateX(-50%) translateY(-50%);
+
+		transition:
+			opacity var(--mod-radio-animation-duration, var(--spectrum-radio-animation-duration)) ease-out,
+			margin var(--mod-radio-animation-duration, var(--spectrum-radio-animation-duration)) ease-out;
+
+		:dir(rtl) & {
+			transform: translateX(50%) translateY(-50%);
+		}
```

I'm wondering if this should be moved to query for the direction of the root class selector rather than looking too far up the DOM tree; in this case it's looking up to root implicitly. Something like:

```suggestion
		.spectrum-Radio:dir(rtl) &::after  {
			transform: translateX(50%) translateY(-50%);
		}
```

---

Great callout, thank you!!